packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -166,6 +194,12 @@ export function buildRelevanceInspector(
     dispatch(enableDebug());
   }
 
+  const warnProductionEnvironment = (flag: string) => {
+    engine.logger.warn(
+      `Flag [ ${flag} ] is now activated. This should *not* be used in any production environment as it negatively impact performance.`
```

Comment:
I think the message is ok for `debug`, but can and should go further for `debugFields`, explaining the correct way to configure fields.

```
The debugFields option negatively impacts query performance. It is helpful for development, but should not be used in production.
Instead, please specify the necessary fields either when instantiating a ResultList controller, or by dispatching a registerFieldsToInclude action.

https://docs.coveo.com/en/headless/latest/reference/controllers/result-list/#resultlistoptions
https://docs.coveo.com/en/headless/latest/reference/actions/field/#registerfieldstoinclude
```

So `warnProductionEnvironment` can be reused by both options, but then concatenating actionable advice for the `debugFields` message.